backend/de.metas.manufacturing/src/main/java/org/eevolution/model/validator/PP_Product_BOM.java

```diff
@@ -58,4 +68,24 @@ public void updateProduct(final I_PP_Product_BOM bom)
 		final ProductId productId = ProductId.ofRepoId(bom.getM_Product_ID());
 		bomService.updateIsBOMFlag(productId);
 	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE },
+			ifColumnsChanged = { I_PP_Product_BOM.COLUMNNAME_PP_Product_BOMVersions_ID, I_PP_Product_BOM.COLUMNNAME_M_Product_ID })
+	public void validateBOMVersions(final I_PP_Product_BOM bom)
+	{
+		final int productId = bom.getM_Product_ID();
+
+		final ProductBOMVersionsId bomVersionsId = ProductBOMVersionsId.ofRepoId(bom.getPP_Product_BOMVersions_ID());
+
+		final I_PP_Product_BOMVersions bomVersions = bomVersionsDAO.getBOMVersions(bomVersionsId);
+
+		if (productId != bomVersions.getM_Product_ID())
+		{
+			throw new AdempiereException("PP_Product_BOM.PP_Product_BOMVersions_ID.M_Product_ID and PP_Product_BOM.M_Product_Id don't match!")
```

Comment:
:speaking_head: EN `The BOM product needs to match the BOM version's product` => DE `Das Produkt der Stückliste muss mit dem Produkt der Stücklistenversion übereinstimmen`